frontend/src/scenes/insights/InsightTabs/TrendTab/Formula.tsx

```diff
@@ -20,7 +22,17 @@ export function Formula({
                 placeholder="e.g. (A + B)/(A - B) * 100"
                 allowClear
                 value={value}
-                onChange={(e) => setValue(e.target.value.toLocaleUpperCase())}
+                onChange={(e) => {
+                    let value = e.target.value.toLocaleUpperCase()
+                    // Only allow typing of allowed characters
+                    value = value
+                        .split('')
+                        .filter((d) => /^[a-zA-Z\ \-\*\^0-9\+\/\(\)]+$/g.test(d))
```

Comment:
You could even restrict to allow adding only letters for the number of graphs currently being showed.